packages/adapter-mongoose/lib/adapter-mongoose.js

```diff
@@ -216,35 +216,22 @@ class MongooseListAdapter extends BaseListAdapter {
         args = mergeWhereClause(args, { id: { $in: ids[0][from.fromField] || [] } });
       }
     }
-    function graphQlQueryToMongoJoinQuery(query) {
-      const _query = {
-        ...query.where,
-        ...mapKeyNames(
-          // Grab all the modifiers
-          pick(query, ['search', 'orderBy', 'skip', 'first']),
-          // and prefix with a dollar symbol so they can be picked out by the
-          // query builder tokeniser
-          key => `$${key}`
-        ),
-      };
-
-      return mapKeys(_query, field => {
-        if (getType(field) !== 'Object' || !field.where) {
-          return field;
-        }
-
-        // recurse on object (ie; relationship) types
-        return graphQlQueryToMongoJoinQuery(field);
-      });
-    }
-
-    let query;
-    try {
-      query = graphQlQueryToMongoJoinQuery(args);
-    } catch (error) {
-      return Promise.reject(error);
```

Comment:
🤔 Why was this catch here? Looking over the original code, I can't see a reason why it might have thrown. But now I'm anxious that we're going to have an edge case we'll no longer be handling safely.

Reply:
Yeah, I looked really closely and I couldn't see any reason for having this error handling pattern here. I can't see any good reason to try, catch, re-throw rather than just letting the original error propagate. Nothing about the parsing looked like it was worth having this there so I decided to clear it out 🤷‍♂

Reply:
The difference here is that we converted the synchronous `Error` into an asynchronous _rejected_ `Promise`, so letting it propagate would change the behaviour of all the calling code when it errors.

Reply:
Mmm, good point, and now I'm seeing a bunch of failures which are perhaps related? Why must programming be so hard!